src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2387,29 +2392,41 @@ static void
 ManageWorkerPool(WorkerPool *workerPool)
 {
 	DistributedExecution *execution = workerPool->distributedExecution;
-	int targetPoolSize = execution->targetPoolSize;
-	int initiatedConnectionCount = list_length(workerPool->sessionList);
-	int activeConnectionCount PG_USED_FOR_ASSERTS_ONLY =
-		workerPool->activeConnectionCount;
-	int idleConnectionCount PG_USED_FOR_ASSERTS_ONLY =
-		workerPool->idleConnectionCount;
-	int failedConnectionCount = workerPool->failedConnectionCount;
-	int readyTaskCount = workerPool->readyTaskCount;
-	int newConnectionCount = 0;
 
-	/* we should always have more (or equal) active connections than idle connections */
-	Assert(activeConnectionCount >= idleConnectionCount);
+	if (HasAnyConnectionFailure(workerPool))
+	{
+		return;
+	}
+	if (ShouldWaitForSlowStart(workerPool))
+	{
+		return;
```

Comment:
Could use a comment to explain why we return:

`/* we wait until a slow start interval has passed before expanding the pool */`